plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/expression/ExpEmbedding.kt

```diff
@@ -376,6 +378,17 @@ data class FieldAccess(val receiver: ExpEmbedding, val field: FieldEmbedding) :
         }
     }
 
+    private fun unfoldingInImpl(ctx: LinearizationContext): Exp {
+        val hierarchyPath = (receiver.type.pretype as? ClassTypeEmbedding)?.details?.hierarchyUnfoldPath(field)
+        val primitiveAccess: Exp = Exp.FieldAccess(receiver.toViper(ctx), field.toViper(), ctx.source.asPosition)
+        if (hierarchyPath == null) return primitiveAccess
+        return hierarchyPath.toList().foldRight(primitiveAccess) { classType, acc ->
+            val predAcc = classType.sharedPredicateAccessInvariant().fillHole(receiver)
+                .pureToViper(toBuiltin = true, ctx.source) as? Exp.PredicateAccess
+            predAcc?.let { Exp.Unfolding(it, acc) } ?: acc
```

Comment:
I would like to have a better understanding of why this doesn't always need to wrap `acc`.  Could you elaborate on it?